File: packages/units/src/length.ts
export const LENGTH_UNITS = [
  "millimeter",
  "centimeter",
  "meter",
  "kilometer",
  "inch",
  "foot",
  "yard",
  "mile"
] as const;

export type LengthUnit = (typeof LENGTH_UNITS)[number];

// How many meters equal 1 unit
const METERS_PER_UNIT: Record<LengthUnit, number> = {
  millimeter: 0.001,
  centimeter: 0.01,
  meter: 1,
  kilometer: 1000,
  inch: 0.0254,
  foot: 0.3048,
  yard: 0.9144,
  mile: 1609.344,
};

export function isLengthUnit(value: string): value is LengthUnit {
  return (LENGTH_UNITS as readonly string[]).includes(value);
}

export function convertLength(value: number, from: LengthUnit, to: LengthUnit): number {
  const meters = value * METERS_PER_UNIT[from];
  return meters / METERS_PER_UNIT[to];
}

File: src/unit_converter/units/length.py
from .linear import LinearUnitCategory

LENGTH_UNITS: tuple[str, ...] = (
    "millimeter",
    "centimeter",
    "meter",
    "kilometer",
    "inch",
    "foot",
    "yard",
    "mile",
)

# How many meters equal 1 unit.
METERS_PER_UNIT: dict[str, float] = {
    "millimeter": 0.001,
    "centimeter": 0.01,
    "meter": 1,
    "kilometer": 1000,
    "inch": 0.0254,
    "foot": 0.3048,
    "yard": 0.9144,
    "mile": 1609.344,
}

_CATEGORY = LinearUnitCategory(LENGTH_UNITS, METERS_PER_UNIT)


def is_length_unit(value: str) -> bool:
    return _CATEGORY.is_unit(value)


def convert_length(value: float, from_unit: str, to_unit: str) -> float:
    return _CATEGORY.convert(value, from_unit, to_unit)
